Website/Controllers/WeBlogImportController.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Web.Http;
+using Sitecore.Jobs;
+using Sitecore.Modules.WeBlog.Import;
+using Sitecore.Services.Core;
+using Sitecore.Services.Infrastructure.Web.Http;
+using Sitecore.StringExtensions;
+
+namespace Sitecore.Modules.WeBlog.Controllers
+{
+    [ServicesController]
+    public class WordPressImportController : ServicesApiController
+    {
+        private string JobHandle { get; set; }
+
+        [HttpPut]
+        [Authorize]
+        public string ImportItems(WordPressImportData data)
+        {
+            var options = new JobOptions("Creating and importing blog", "WeBlog", Context.Site.Name, this, "ImportBlog", new[] { data });
+            var job = JobManager.Start(options);
+            job.Status.Total = 0;
+            JobHandle = job.Handle.ToString();
+            return JobHandle;
+        }
+
+        private void ImportBlog(WordPressImportData data)
+        {
+            var master = Sitecore.Configuration.Factory.GetDatabase("master");
```

Comment:
Rather than using a hard coded DB name ("master"), instead we should ask which DB to import into and list all available DBs in a droplist. Although not common, it might be that the master DB has had it's name changed.